json-fleece-openapi3/src/Fleece/OpenApi3.hs

```diff
@@ -1623,3 +1624,129 @@ mkAdditionalPropertiesSchema raiseError schemaKey mkInlineItemSchema mbAdditiona
+            propB = OA._discriminatorPropertyName discB
+          in
+            if propA == propB
+              then
+                pure
+                  . Just
+                  $ OA.Discriminator
+                    { OA._discriminatorPropertyName = propA
+                    , OA._discriminatorMapping =
+                        OA._discriminatorMapping discA <> OA._discriminatorMapping discB
+                    }
+              else
+                lift . CGU.codeGenError $
+                  T.unpack typeNameText <> ": Cannot combine discriminators with different property names."
+
+  format <-
+    let
+      schemaFormatA = OA._schemaFormat schemaA
+      schemaFormatB = OA._schemaFormat schemaB
+    in
+      if fromMaybe True $ liftA2 (==) schemaFormatA schemaFormatB
+        then pure (schemaFormatA <|> schemaFormatB)
+        else
+          lift
+            . CGU.codeGenError
+            . unwords
+            $ [ T.unpack typeNameText <> ":"
+              , "Schema formats do not match for all items in allOf array: got"
+              , maybe "<no format>" show schemaFormatA
+              , "and"
+              , maybe "<no format>" show schemaFormatB <> "."
+              ]
+
+  items <-
+    let
+      schemaItemsA = OA._schemaItems schemaA
+      schemaItemsB = OA._schemaItems schemaB
+    in
+      if all isJust [schemaItemsA, schemaItemsB]
+        then
+          lift
+            . CGU.codeGenError
+            $ T.unpack typeNameText <> ": Multiple arrays in allOf schemas are not currently supported."
+        else
+          traverse (inlineAllOfItems (typeNameText <> ".Items")) (OA._schemaItems schemaA <|> OA._schemaItems schemaB)
+
+  schemaType <-
+    let
+      schemaTypeA = OA._schemaType schemaA
+      schemaTypeB = OA._schemaType schemaB
+    in
+      if fromMaybe True $ liftA2 (==) schemaTypeA schemaTypeB
+        then pure (schemaTypeA <|> schemaTypeB)
+        else
+          lift
+            . CGU.codeGenError
+            . unwords
+            $ [ T.unpack typeNameText <> ":"
+              , "Schemas in allOf array must all be the same type, but got"
+              , maybe "<no type>" show schemaTypeA
+              , "and"
+              , maybe "<no type>" show schemaTypeB <> "."
+              ]
+
+  let
+    traverseInline = (traverse . traverse) (inlineAllOfReferenced typeNameText)
+    traverseProps =
+      IOHM.unorderedTraverseWithKey
+        (\k v -> inlineAllOfReferenced (typeNameText <> "." <> k) v)
+
+  -- TODO: This prefers properties from the left map. May not match OpenAPI spec.
+  props <- traverseProps $ OA._schemaProperties schemaA <> OA._schemaProperties schemaB
+  allOf <- traverseInline $ OA._schemaAllOf schemaA <> OA._schemaAllOf schemaB
+  anyOf <- traverseInline $ OA._schemaAnyOf schemaA <> OA._schemaAnyOf schemaB
+  oneOf <- traverseInline $ OA._schemaOneOf schemaA <> OA._schemaOneOf schemaB
+
+  pure $
+    (schemaA <> schemaB)
```

Comment:
Why use `<>` here at all? If we're implementing our own append logic it's probably more clear to construct a new schema from scratch and make every field have our own explicit behavior.